### PR TITLE
Pull Request for Issue1892: Adding simple control view

### PR DESCRIPTION
**Pull request by helfrij**
Added additional option to BioXASControlEditor context menu to view control properties. This causes a widget to pop up with basic AMControl information.

### CODE REVIEW

**Comment by helfrij** [comment]
Hey @XuetaoLiu-CLS, would you review this pull request?

**Comment by XuetaoLiu-CLS** [comment]
Looks good. 

Time Estimates: 10 minutes

File: source/ui/BioXAS/BioXASControlEditor.cpp [hunk 1]
```diff
@@ -365,6 +379,15 @@ void BioXASControlEditor::onCalibrateActionTriggered()
 	}
 }
 
+void BioXASControlEditor::onPropertiesActionTriggered()
+{
+	if (control_) {
+		AMControlView *view = new AMControlView(control_);
+		view->setWindowTitle(QString("%1 properties").arg(title()));
+		view->show();
```
Comment (dretrex) [inline source/ui/BioXAS/BioXASControlEditor.cpp]: You may want to add `view->raise();` to guarantee that it shows up.  

Also, this is a memory leak from what I can tell as nothing keeps track of this pointer.

File: source/ui/beamline/AMControlView.cpp [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+		refresh();
+
+		emit controlChanged(control_);
+	}
+}
+
+void AMControlView::updateNameLabel()
+{
+	QString name;
```
Comment (dretrex) [inline source/ui/beamline/AMControlView.cpp]: Although not strictly necessary, I'd initialize these QStrings to `""`.

**Comment by dretrex** [comment]
Only a couple comments.
Time Estimate: 5 minutes

File: source/ui/BioXAS/BioXASControlEditor.cpp (hunk 1 above)
Comment (helfrij) [inline source/ui/BioXAS/BioXASControlEditor.cpp]: You're right that BioXASControlEditor doesn't keep track. Would a better solution be to listen for a closed() type signal and explicitly delete this widget then?

Comment (helfrij) [inline source/ui/BioXAS/BioXASControlEditor.cpp]: The view is now deleted explicitly when closed.

**Comment by helfrij** [comment]
@dretrex 
Let me know if there is anything else to look into here.

**Comment by dretrex** [comment]
Looks good!

**Pull request merged**